p2p_wallet/Scenes/Main/SendToken/Subscenes/ChooseRecipientAndNetwork/SelectAddress/Subviews/RecipientsCollectionView/SendToken.ChooseRecipientAndNetwork.SelectAddress.RecipientsListViewModel.swift

```diff
@@ -78,37 +79,32 @@ extension SendToken.ChooseRecipientAndNetwork.SelectAddress {
         }
 
         private func findAddressInSolanaNetwork(address: String) -> Single<[SendToken.Recipient]> {
-            Single<String?>.async { [weak self] in
-                guard let self = self else { throw NameServiceError.unknown }
-                return try await self.nameService
-                    .getName(address)
-            }
-            .flatMap { [weak self] name -> Single<(String?, Bool)> in
-                guard let self = self, name == nil else { return .just((name, false)) }
-                // check funds
-                return Single.async {
-                    try await self.solanaAPIClient.checkAccountValidation(account: address)
+            Observable<String?>.just(nil)
+                .asSingle()
+                .flatMap { [weak self] name -> Single<(String?, Bool)> in
+                    guard let self = self, name == nil else { return .just((name, false)) }
+                    // check funds
+                    return Single.async {
+                        try await self.solanaAPIClient.checkAccountValidation(account: address)
+                    }
+                    .catchAndReturn(false)
+                    .map { (name, !$0) }
```

Comment:
The expression is too complicated, you can just use something like this:
```swift 
Single<Bool>.async { [weak self] in
    try? await self?.solanaAPIClient.checkAccountValidation(account: address) ?? false // won't return error because of "try?"
}
.map {
    [
          .init(
              address: address,
              name: nil,
              hasNoFunds: $0
          ),
    ]
}
```